src/Aspire.Hosting.AWS/CloudFormation/CloudFormationProvisioner.cs

```diff
@@ -248,13 +256,23 @@ private async Task ProcessCloudFormationTemplateResourceAsync(CancellationToken
         }
     }
 
+    private static bool IsStackInErrorState(Stack stack)
+    {
+        if(stack.StackStatus.ToString(CultureInfo.InvariantCulture).EndsWith("FAILED", StringComparison.OrdinalIgnoreCase) || stack.StackStatus.ToString(CultureInfo.InvariantCulture).EndsWith("ROLLBACK_COMPLETE", StringComparison.OrdinalIgnoreCase))
+        {
+            return true;
+        }
+
+        return false;
+    }
+
     private async Task PublishCloudFormationUpdateStateAsync(CloudFormationResource resource, CustomResourceSnapshot state)
     {
-        await notificationService.PublishUpdateAsync(resource, state).ConfigureAwait(false);
+        await notificationService.PublishUpdateAsync(resource, (s) => state).ConfigureAwait(false);
```

Comment:
This should really be doing state mutations instead of passing in new state. The pattern is `s => s with { }` state updates are immutable snapshots

Reply:
Fixed